HW_3.py

```diff
@@ -0,0 +1,163 @@
+#Task 1
+
+import requests
+from bs4 import BeautifulSoup
+from collections import defaultdict
+import matplotlib.pyplot as plt
+import pandas as pd
+import re
+
+response = requests.get('https://www.imdb.com/chart/top/?ref_=nv_mp_mv250')
+soup = BeautifulSoup(response.text, 'html.parser')
+table = soup.find('tbody', {'class': 'lister-list'})
+rows = table.findAll('tr')
+movies = []
+director=[]
+
+for row in rows:
+    title_column = row.find('td', {'class': 'titleColumn'})
+    name = title_column.a.text
+    year = title_column.span.text.strip('()')
+    rating = float(row.find('td', {'class': 'ratingColumn 
+imdbRating'}).text.strip())
+    n_reviews = int(row.find('td', {'class': 'ratingColumn 
+imdbRating'}).find('strong').attrs['title'].split()[3].replace(',', ''))
+    director = (re.search(r'title="([^"]+)"', str(title_column)).group(1))
+    director = director.split(' (dir.)')[0]
+    rank = int(row.find('td', {'class': 
+'titleColumn'}).text.strip().split('.')[0])
+    movie = {'name': name, 'rank': rank, 'year': year, 'rating': rating, 
+'n_reviews': n_reviews, 'director': director}
+    movies.append(movie)
+top_n_reviews = sorted(movies, key=lambda x: x['n_reviews'], 
+reverse=True)[:4]
+for movie in top_n_reviews:
+    print(f"{movie['name']}: {movie['n_reviews']} reviews")
+year_ratings = defaultdict(list)
+top_years = []
+for movie in movies:
+    year_ratings[movie['year']].append(movie['rating'])
+
+for year, ratings in year_ratings.items():
+    avg_rating = sum(ratings) / len(ratings)
+    top_years.append((year, avg_rating))
+
+top_years = sorted(top_years, key=lambda x: x[1], reverse=True)[:4]
+for year, avg_rating in top_years:
+    print(f"{year}: {avg_rating:.2f}")
+director_counts = defaultdict(int)
+for movie in movies:
+    director_counts[movie['director']] += 1
+    
+director_counts = {k: v for k, v in director_counts.items() if v > 2}
+director_counts = {k: v for k, v in sorted(director_counts.items(), 
+key=lambda x: x[1], reverse=True)}
+plt.bar(director_counts.keys(), director_counts.values())
+plt.xticks(rotation=90)
+plt.xlabel('Режиссёр')
+plt.ylabel('Количество фильмов в топ-250 IMDb')
+plt.show()
+pd.DataFrame(movies).to_csv('top250_full.csv', index=False)
+
+#Task 2
+
+import sys
+import time
+import os
+import io
+import datetime
+import requests
+from dotenv import load_dotenv
+
+load_dotenv()
+TG_API_TOKEN = os.environ.get('TG_API_TOKEN')
+def telegram_logger(chat_id):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+        
+            start_time = time.monotonic()
```

Comment:
При вычитании `datetime` автоматически получается `timedelta`
```suggestion
            start_time = datetime.now()
```